src/main/java/com/semmle/jira/addon/LgtmServlet.java

```diff
@@ -150,27 +154,42 @@ void createIssue(
     issueInputParameters.setIssueTypeId(JiraUtils.getLgtmIssueType().getId());
 
     issueInputParameters.addProperty(Constants.LGTM_PAYLOAD_PROPERTY, rawRequest);
+
     issueInputParameters.setApplyDefaultValuesWhenParameterNotProvided(true);
 
     IssueService.CreateValidationResult createValidationResult =
         ComponentAccessor.getIssueService().validateCreate(config.getUser(), issueInputParameters);
 
     if (createValidationResult.getErrorCollection().hasAnyErrors()) {
       writeErrors(createValidationResult, resp);
-    } else {
-      IssueService.IssueResult issueResult =
-          ComponentAccessor.getIssueService().create(config.getUser(), createValidationResult);
+      return;
+    }
+
```

Comment:
Could you tighten the scope of the try-catch a bit, like:
```

CustomField field;
try {
  field = JiraUtils.getConfigKeyCustomField();
} catch (CustomFieldRetrievalException e) {
      sendError(
          resp,
          HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
          "Retrieval of custom field for config key failed.");
      return;
 }
createValidationResult
          .getIssue()
          .setCustomFieldValue(field, config.getKey());
```
That way it is more clear that we only handle errors from the `getConfigKeyCustomField` method.